### ISSUE TITLE
Vite-dev-server PluginConfig typescript errors

### ISSUE THREAD

**Issue by eratio08** [cypress-io/cypress#17648]
### Current behavior

I want to test the component testing capabilities of cypress using vite-dev-server and vue. When I try to wire up the setup in the `plugins/index.ts` I get the following type error:

```shell
The expected type comes from property 'options' which is declared here on type 'StartDevServerOptions'

Type 'DevServerOptions' is not assignable to type 'Options'.
  Types of property 'config' are incompatible.
    Type 'ResolvedConfigOptions & RuntimeConfigOptions' is not assignable to type 'Record<string, string>'.
      Index signature is missing in type 'ResolvedConfigOptions & RuntimeConfigOptions'.ts(2322)
```

### Desired behavior

No typescript errors

### Test code to reproduce

```typescript
import { startDevServer } from '@cypress/vite-dev-server/dist' // I need to import dist because the module root has not .d.ts

const pluginConfigHook: Cypress.PluginConfig = (on, config) => {
	if (config.testingType === 'component') {
		on('dev-server:start', async (options) => startDevServer({ options }))
	}

	return config
}
```

### Cypress Version

8.2.0

### Other

"@cypress/vite-dev-server": "^2.0.5"

**Comment by JessicaSachs** [comment]
CC @elevatebart can you pick this up tomorrow?

**Comment by elevatebart** [comment]
The type error is fixed in `@cypress/vite-dev-server@2.0.7` so if you update you should not see the error anymore.
I still need to publish index.d.ts to npm in the adjacent Pull Request #17786

**Comment by JessicaSachs** [comment]
Approved. Should go out momentarily.
